src/Libraries/Microsoft.Extensions.AI.Evaluation.NLP/BLEU/RationalNumber.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+namespace Microsoft.Extensions.AI.Evaluation.NLP.BLEU;
+
+internal readonly struct RationalNumber : IEquatable<RationalNumber>
+{
+    public RationalNumber(int numerator, int denominator)
+    {
+        if (denominator == 0)
+        {
+            throw new DivideByZeroException("Denominator cannot be zero.");
+        }
+
+        Numerator = numerator;
+        Denominator = denominator;
+    }
+
+    public int Numerator { get; }
+    public int Denominator { get; }
+
+    public double ToDouble() => (double)Numerator / Denominator;
+
+    public override string ToString() => $"{Numerator}/{Denominator}";
```

Comment:
Same comment about overriding `ToString()` v/s `[DebuggerDisplay]`. I think the latter may be preferrable if this is solely for debugging purposes.